manifests/init.pp

```diff
@@ -0,0 +1,166 @@
+# Sets up rkhunter to be run once per day
+#
+# @param minute_upd   Cron minute for update check
+# @param hour_upd     Cron hour for update check
+# @param monthday_upd Cron monthday for update check
+# @param month_upd    Cron month for update check
+# @param weekday_upd  Cron weekday for update check
+# @param minute       Cron minute
+# @param hour         Cron hour
+# @param monthday     Cron monthday
+# @param month        Cron month
+# @param weekday      Cron weekday
+#
+# @param package_ensure                The ensure status of packages to be managed
+#
+# @param rkhunter_conf_file            The path to the conf file
+#
+# @param rkhunter_conf_file_template   Template to use for the conf file
+#
+# @param check_for_updates             Check internet for definition updates
+#
+# @author https://github.com/simp/pupmod-simp-rkhunter/graphs/contributors
+#
+class rkhunter (
+  String[1]         $minute_upd                      = '0',
+  String[1]         $hour_upd                        = '0',
+  String[1]         $monthday_upd                    = '*',
+  String[1]         $month_upd                       = '*',
+  String[1]         $weekday_upd                     = '*',
+  String[1]         $minute                          = '10',
+  String[1]         $hour                            = '0',
+  String[1]         $monthday                        = '*',
+  String[1]         $month                           = '*',
+  String[1]         $weekday                         = '*',
+  String[1]         $package_ensure                  = 'latest',
+  String[1]  $rkhunter_conf_file                    = '/etc/rkhunter.conf',
+  String[1]  $rkhunter_conf_file_template           = 'rkhunter/rkhunter-conf.erb',
+  Optional[Boolean]    $check_for_updates            = undef,
+############start of rkhunter.conf options
+  Optional[Array[String[1]]]  $allowdevfile                      = undef,
+  Optional[Array[String[1]]]  $allowhiddendir                    = undef,
+  Optional[Array[String[1]]]  $allowhiddenfile                   = undef,
+  Optional[Array[String[1]]]  $allowipcpid                       = undef,
```

Comment:
Yes, however rkhunter throws a wrench in that because it allows you to have it be:
ALLOWIPCPID:435 6486 68431 684613

which necessitates a string.

Reply:
@sharkbruhaha Not following this. If you restrict it to `Array[Integer[1]]`, you can format it however you like in the template.

Unless there's some reason that we can't do this, we don't always expose every possible formatting combination.

Reply:
in puppet terms, it allows for:
`ALLOWIPCPID => ['435 6486 68431 684613',4694]`
which are then resolved to:
`ALLOWIPCPID=435 6486 68431 684613
ALLOWIPCPID=4694`
Both are allowed and all are read in by rkhunter when it looks for `ALLOWIPCPID`

Reply:
if we just want to say screw that noise, in favour of having some ease of parameter checking for the user then, sure we can have `Array[Integer[1]]`, but if a user that is actually reading the rkhunter docs comes along and tries to do a one-liner, then it is going to be problematic. That is why I was shying away from parameter checking for this because almost everything in here can be a space separated list in addition to an array. It just gets really messy trying to idiot proof it, when this is a program that necessitates the user reading the docs for it if they are going to start mucking with much of its config.

Reply:
@sharkbruhaha Still agree with @lnemsick-simp. Users need to read the documentation in the module to use the module.